core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js

```diff
@@ -138,18 +145,54 @@ oppia.factory('SimpleEditorManagerService', [
           data.questionList.isEmpty() ?
           SimpleEditorShimService.getInitStateName() :
           data.questionList.getLastQuestion().getDestinationStateName());
-
         SimpleEditorShimService.saveInteractionId(
-          lastStateName, DEFAULT_INTERACTION.ID);
+          lastStateName, 'MultipleChoiceInput');
         SimpleEditorShimService.saveCustomizationArgs(
-          lastStateName, DEFAULT_INTERACTION.CUSTOMIZATION_ARGS);
+          lastStateName,
+          DEFAULT_INTERACTION_PROPERTIES.MultipleChoiceInput.CUSTOMIZATION_ARGS
+        );
         SimpleEditorShimService.saveDefaultOutcome(
           lastStateName, OutcomeObjectFactory.createEmpty(lastStateName));
 
         var stateData = SimpleEditorShimService.getState(lastStateName);
         data.questionList.addQuestion(QuestionObjectFactory.create(
           lastStateName, stateData.interaction, ''));
       },
+      changeQuestion: function(newQuestionType, index) {
+        var currentStateName = data.questionList.getAllStateNames()[index];
+        var nextStateName = data.questionList.getAllStateNames()[index + 1];
+        var questionCount = data.questionList.getQuestionCount();
+        var currentInteractionId = (
+          SimpleEditorShimService.getInteractionId(currentStateName));
+        var doesLastQuestionHaveAnswerGroups = (
+          QuestionListObjectFactory.doesLastQuestionHaveAnswerGroups);
+
+        // Update Question Type If interactionId is not same.
+        if (newQuestionType !== currentInteractionId) {
+          var newAnswerGroups = [];
+          SimpleEditorShimService.saveInteractionId(
+            currentStateName, newQuestionType);
+          SimpleEditorShimService.saveCustomizationArgs(
+            currentStateName,
+            DEFAULT_INTERACTION_PROPERTIES[newQuestionType].CUSTOMIZATION_ARGS);
+          newAnswerGroups.push(AnswerGroupObjectFactory.createNew([
+            RuleObjectFactory.createNew(
+              DEFAULT_INTERACTION_PROPERTIES[newQuestionType].
+                FIRST_ANSWER_GROUP_RULE.type,
+              DEFAULT_INTERACTION_PROPERTIES[newQuestionType].
+                FIRST_ANSWER_GROUP_RULE.value
+            )
+          ], OutcomeObjectFactory.createEmpty(nextStateName), false));
+
+          if (doesLastQuestionHaveAnswerGroups && index !== questionCount - 1) {
```

Comment:
Ah thanks. I just saw your earlier reply, but I think it's still not clear to a developer who is reading the code.

Here's the reply you wrote:

> A last state gets added only when answers is selected, this to prevent the data getting corrupted when user add questions and then change it without selecting answer.

but I still don't understand it. Can you elaborate? How does it apply to the scenario I mentioned above?